### ISSUE TITLE
base.UnmarshalRoutes fails on non-addpath default routes

### ISSUE THREAD

**Issue by taktv6** [sbezverk/gobmp#202]
The function UnmarshalRoutes (https://github.com/sbezverk/gobmp/blob/master/pkg/base/route.go#L39) fails to decode non-addpath NLRIs.
Here is a test that shows the problem:
```func TestUnmarshalBaseNLRI(t *testing.T) {
	tests := []struct {
		name   string
		input  []byte
		expect []Route
	}{
		{
			name:  "fail",
			input: []byte{0, 8, 10, 24, 10, 0, 0, 24, 20, 0, 0},
			expect: []Route{
				{
					Length: 0x0,
					Prefix: []byte{},
				},
				{
					Length: 0x08,
					Prefix: []byte{10},
				},
				{
					Length: 24,
					Prefix: []byte{10, 0, 0},
				},
				{
					Length: 24,
					Prefix: []byte{20, 0, 0},
				},
			},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := UnmarshalRoutes(tt.input)
			if err != nil {
				t.Fatalf("test failed with error: %+v", err)
			}
			if !reflect.DeepEqual(tt.expect, got) {
				t.Fatalf("test failed as expected nlri %+v does not match actual nlri %+v", tt.expect, got)
			}
		})
	}
}```

Failure:
```Running tool: /home/takt/software/go/bin/go test -timeout 30s -run ^TestUnmarshalBaseNLRI$ github.com/sbezverk/gobmp/pkg/base

--- FAIL: TestUnmarshalBaseNLRI (0.00s)
    --- FAIL: TestUnmarshalBaseNLRI/fail (0.00s)
        /home/takt/git/src/github.com/taktv6/gobmp/pkg/base/route_test.go:102: test failed as expected nlri [{PathID:0 Length:0 Label:[] RD:<nil> Prefix:[]} {PathID:0 Length:8 Label:[] RD:<nil> Prefix:[10]} {PathID:0 Length:24 Label:[] RD:<nil> Prefix:[10 0 0]} {PathID:0 Length:24 Label:[] RD:<nil> Prefix:[20 0 0]}] does not match actual nlri [{PathID:526872 Length:10 Label:[] RD:<nil> Prefix:[0 0]} {PathID:0 Length:24 Label:[] RD:<nil> Prefix:[20 0 0]}]
FAIL
FAIL	github.com/sbezverk/gobmp/pkg/base	0.003s
FAIL```

The NLRI encoding can not be detected reliably by the NLRI's content.
What is needed is a calculation of the negotiated capabilities from the BGP OPEN messages from the BMP Peer Up Notifications.
With that capability information is then to be decided how the NLRI must be decoded.

**Comment by sbezverk** [comment]
@taktv6  thank you for filing the issue. Could you please run gobmp with --v=9 in the test bed where bgp speakers are not add-path capable and collect the hex for updates carrying NLRI without path-id. It will help to test the logic of the fix. Thank you.

**Comment by taktv6** [comment]
I just realized I posted an incomplete test. So I updated that in the code above.
In the test you see a hex / dec NLRI series and the expected result. From the test output one can see how a path ID is decoded, while there is not add-path involved here. The only solution to this is to have a bool flag into the unmarshal function that indicates if add-path was enabled on the BGP session that the reported UPDATE messages was received on. To know this it is necessary to inspect the OPEN messages that the peers have exchanged (that are reported in the peer up notification) to find out if add-path NLRI encoding is used or not. There is no way to fix this by only changing the unmarshal function.
